x-pack/plugins/observability_solution/entity_manager/server/lib/entities/install_entity_definition.ts

```diff
@@ -46,110 +53,69 @@ export interface InstallDefinitionParams {
-    indexTemplates: {
-      history: false,
-      latest: false,
-    },
-  };
-
   try {
-    logger.debug(() => `Installing definition ${JSON.stringify(definition)}`);
-
     validateDefinitionCanCreateValidTransformIds(definition);
 
-    const entityDefinition = await saveEntityDefinition(soClient, definition);
-    installState.definition = true;
-
-    // install scoped index template
-    await upsertTemplate({
-      esClient,
-      logger,
-      template: getEntitiesHistoryIndexTemplateConfig(definition),
-    });
-    installState.indexTemplates.history = true;
-    await upsertTemplate({
-      esClient,
-      logger,
-      template: getEntitiesLatestIndexTemplateConfig(definition),
-    });
-    installState.indexTemplates.latest = true;
-
-    // install ingest pipelines
-    logger.debug(`Installing ingest pipelines for definition ${definition.id}`);
-    await createAndInstallHistoryIngestPipeline(esClient, entityDefinition, logger);
-    installState.ingestPipelines.history = true;
-    await createAndInstallLatestIngestPipeline(esClient, entityDefinition, logger);
-    installState.ingestPipelines.latest = true;
-
-    // install transforms
-    logger.debug(`Installing transforms for definition ${definition.id}`);
-    await createAndInstallHistoryTransform(esClient, entityDefinition, logger);
-    installState.transforms.history = true;
-    if (isBackfillEnabled(entityDefinition)) {
-      await createAndInstallHistoryBackfillTransform(esClient, entityDefinition, logger);
-      installState.transforms.backfill = true;
+    if (await entityDefinitionExists(soClient, definition.id)) {
+      throw new EntityIdConflict(
+        `Entity definition with [${definition.id}] already exists.`,
+        definition
+      );
     }
-    await createAndInstallLatestTransform(esClient, entityDefinition, logger);
-    installState.transforms.latest = true;
 
-    return entityDefinition;
+    return await install({ esClient, soClient, definition, logger });
   } catch (e) {
     logger.error(`Failed to install entity definition ${definition.id}: ${e}`);
-    // Clean up anything that was successful.
-    if (installState.definition) {
-      await deleteEntityDefinition(soClient, definition, logger);
-    }
 
-    if (installState.ingestPipelines.history) {
-      await deleteHistoryIngestPipeline(esClient, definition, logger);
-    }
-    if (installState.ingestPipelines.latest) {
-      await deleteLatestIngestPipeline(esClient, definition, logger);
-    }
+    await deleteEntityDefinition(soClient, definition).catch((err) => {
+      if (err instanceof EntityDefinitionNotFound) {
+        return;
+      }
+      throw err;
+    });
 
-    if (installState.transforms.history) {
-      await stopAndDeleteHistoryTransform(esClient, definition, logger);
-    }
+    await Promise.all([
+      stopAndDeleteHistoryTransform(esClient, definition, logger),
+      stopAndDeleteHistoryBackfillTransform(esClient, definition, logger),
```

Comment:
Is it needed here to check if the backfill is enabled?

Reply:
it's a noop if the transform does not exist but we can save a network call

Reply:
see the rationale here https://github.com/elastic/kibana/pull/188351/files#r1703843563

Reply:
That's fine then